rvv-intrinsic-rfc.md

```diff
@@ -726,3 +727,32 @@ vuint8m1_t vreinterpret_u8 (vint8m1_t src);
 // x for scalar, xu for unsigned scalar, and f for float.
 vint16m1_t vfcvt_x (vfloat16m1_t src);
 ```
+
+## Switching Vtype and Keep same VL after vsetvl instruction<a name="switching-vtype"></a>
+
+Compiler should guarantee the correctness of vtype setting after vsetvl instruction. For example considering the widening multiply example as below.
+
+```
+vl = vsetvl_e16m4(n);
+vfloat16m4_t vx = vle16_v_f16m4(ptr_x);
+// vsetvl_e32m8(vl); // keep the same vl without change vtype manually
+vfloat32m8_t vy = vle32_v_f32m8(ptr_y);
+// vsetvl_e16m4(vl); // keep the same vl without change vtype manually
+vfwmacc_vf_f32m8(vy, 2.0, vx);
+// vsetvl_e32m8(vl); // keep the same vl without change vtype manually
+vse32_v_f32m8(ptr_y, vy);
+```
+
+```
+vl = vsetvl_e16m4(n);
+vfloat16m4_t vx = vle16_v_f16m4(ptr_x, vl);
+// vsetvl_e32m8(vl); // change vtype manually
+vfloat32m8_t vy = vle32_v_f32m8(ptr_y, vl);
+// vsetvl_e16m4(vl); // change vtype manually
+vfwmacc_vf_f32m8(vy, 2.0, vx, vl);
+// vsetvl_e32m8(vl); // change vtype manually
+vse32_v_f32m8(ptr_y, vy, vl);
+
+This example has a `vl` computed from `vsetvle16m4_t`, and changing the type to `vfloat32m8_t` in the middle.
+With compiler's helping, users don't need to change vtype manually because `vfloat16m4_t` and `vfloat32m8_t` have the exact same number of elements (same `SEW`/`LMUL` ratio).
+Noted that when changing vtype with different `SEW`/`LMUL` ratio in a loop, the result is undefined.
```

Comment:
It seems will result an illegal-instruction exception more than undefined.
https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#61-vsetvlivsetvl-instructions
```
...
Use of the instruction with a new SEW/LMUL ratio that would result in a
change of VLMAX is reserved. Implementations may set vill in this case.
...
```

Once `vill` is raised:
https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#334-vector-type-illegal-vill
```
If the vill bit is set, then any attempt to execute a vector instruction that depends
upon vtype will raise an illegal-instruction exception.
```

> Noted that when changing vtype with different `SEW`/`LMUL` ratio in a loop, the result is undefined.

I worry about this wording is too strong, for example, with the reduction operation, we might change the vtype with VLMAX change within a loop, and according this statement the result is undefined since we've changing `SEW`/`LMUL` ratio within a loop.

Ref:
https://github.com/sifive/sifive-dsp-internal/blob/master/DSP/DSP-rvv/Source/FilteringFunctions/riscv_lms_f32.c#L122

Reply:
I changed the description, hope it will be more clear.